[Mobile] Improve the design of shop page messages (closed shop, private shop and select OC)

luisramos0 opened the pull request:
#### What? Why?

Closes #4772 and #5216 and #4774
Also fixes the last point from #4573 (and so it closes the issue):
- The shop closed message will display below both the 'home' and 'shop' tabs on mobile, tablet, and desktop (see designs below)

In both private shop and closed shop messages: removed border of message and paddings, add background color, change link color and split message in several lines.
In closed shop, added darker background with new warning sign.
In the closed shop, implement 5216, show either the default message OR the user defined one.
From 4753, show closed shop message on home tab as well.
Re 4774, added the message to select an OC if there are multiple OCs.

#### What should we test?
See the issues and the screenshots:
- for 4772 [here](https://github.com/openfoodfoundation/openfoodnetwork/issues/4772#issuecomment-615932600)
- for 5116 [here](https://github.com/openfoodfoundation/openfoodnetwork/issues/5216#issuecomment-617245064).
- for 4774 [here](https://github.com/openfoodfoundation/openfoodnetwork/issues/4774#issuecomment-622933666)

#### Release notes
Changelog Category: Changed
Improved design of closed shopfront messages (when login is required, when shop is for customers only and when shop is simply closed).
Default closed shop message will not show if there is a user defined message.
Closed shop message will also show on the Home tab if shop is closed.
Added message to the body of the page for user to select Order Cycle when there are multiple Order cycles available.

luisramos0 commented on app/assets/stylesheets/darkswarm/shop_tabs.css.scss:
```diff
@@ -76,20 +76,17 @@
 
   .page-view {
-    margin-bottom: 5em;
+    border-bottom: 5em solid $grey-080;
```
the upper part of the background below this image was margin (cant change background color of margin) now it's border (can change background color of border)
![image](https://user-images.githubusercontent.com/1640378/79683595-2980e200-8223-11ea-891d-e96b59878d15.png)

luisramos0 commented on app/assets/stylesheets/darkswarm/shop_tabs.css.scss:
```diff
@@ -76,20 +76,17 @@
     background: none;
+    background-color: $grey-080;
     border: none;
+    padding-bottom: 5em;
```
the upper part of the background below this image was margin (doesnt take the background color of the content and we cannot change background color of margin) now it's padding (takes the background color of the content)
![image](https://user-images.githubusercontent.com/1640378/79683595-2980e200-8223-11ea-891d-e96b59878d15.png)

luisramos0 commented on app/assets/stylesheets/darkswarm/shop.css.scss:
```diff
@@ -125,21 +125,44 @@
+      display: block;
+      position: relative;
+      width: 23px;
+    }
```
I saw the warning sign done with css in zeplin so I gave it a try. It's not a beauty but acceptable?
![image](https://user-images.githubusercontent.com/1640378/79856172-e4e07c80-83c3-11ea-9d17-8984527069c7.png)

sauloperez replied:
isn't there an icon from font-awesome that we could use? it seems a bit more maintainable to me given we already import it.

luisramos0 replied:
I dont think fontawesome is used in the frontoffice...
I spent one more hour on this without success so I am leaving as it is...

sauloperez replied:
Who knows what mess we might have between front/backoffice assets...

luisramos0 commented on app/views/shop/_closed_shop.html.haml:
```diff
@@ -0,0 +1,21 @@
+.row.closed_shop_header
+  .small-12.columns
+    .content{ "darker-background" => true }
```
First I repeated some conditionals at the page-view element level to set the background darker but then I tried this directive approach and it looks shorter (and faster without duplicate conditionals).
Is there a better way of doing this I may be missing?

sauloperez commented on app/assets/stylesheets/darkswarm/shop.css.scss:
```diff
@@ -125,21 +125,44 @@
-  .shopfront_closed_message {
-    margin: 2em 0em;
+    p {
+      line-height: 2.5rem;
```
this is a quite large `line-height` for a paragraph are we sure it's not margin what we need here?

sauloperez replied:
ping @luisramos0

luisramos0 replied:
yes, you are correct, I fixed it :+1:

sauloperez commented on app/assets/javascripts/darkswarm/directives/darker_background.js.coffee:
```diff
@@ -0,0 +1,4 @@
+Darkswarm.directive "darkerBackground", ->
+  restrict: "A"
+  link: (scope, elm, attr)->
+    elm.closest('.page-view').toggleClass("with-darker-background", true)
```
so this directive makes the background color toggleable? I'm not sure I'm following.

luisramos0 replied:
yes, that's it, it toggles the background color of the parent page-view element. This avoids having to repeat the conditionals when we define the parent element on the server (i.e. if tab is shop and shop is closed OR...)

Matt-Yorkley replied:
If there's any other way we can set this background colour with just CSS (without adding this extra directive) I think I'd favour that. I'll have a quick look at it now...

Matt-Yorkley replied:
Ok, it's pretty tricky. I think it can be done, but it might take a lot of refactoring and moving everything around :see_no_evil: 

I guess this option is simpler :+1:

luisramos0 replied:
yes, I tried the other way with conditionals on the server side! I then moved to this one and thought this was better.

Matt-Yorkley commented:
Can we get a screengrab or something? I've been using **Peek**, it's really nice: https://www.fossmint.com/peek-gif-screen-recorder-for-linux/

luisramos0 commented:
There are links to screenshots on the description. This is not so good to screen grab the action as each situation requires specific setup on the backoffice.

luisramos0 commented:
![Peek 2020-04-29 10-44](https://user-images.githubusercontent.com/1640378/80582350-74aaaa00-8a06-11ea-980f-774db2f02d9d.gif)

Matt-Yorkley commented:
I'm also getting missing translations on the customer login section:

![Screenshot from 2020-05-01 10-49-39](https://user-images.githubusercontent.com/9029026/80794208-a56f1880-8b99-11ea-99f1-55080522a3a0.png)

Matt-Yorkley commented:
Also, we can get _exactly_ the same shade of grey if we use `rgba(0,0,0,0.05)` here instead of `#f2f2f2f2`, but it means the drop-shadow on the tabs above is still visible instead of obscured:

![Screenshot from 2020-05-01 11-05-48](https://user-images.githubusercontent.com/9029026/80795112-e1a37880-8b9b-11ea-9e08-c2e89101a3d8.png)
![Screenshot from 2020-05-01 11-05-56](https://user-images.githubusercontent.com/9029026/80795113-e2d4a580-8b9b-11ea-9c35-9fdd026c45b0.png)

luisramos0 commented:
I rebased this just to get rid of the trackers table (dropped in master).

luisramos0 commented:
ok guys, thanks a lot for your feedback.

Matt, thanks for spotting all the problems!
- I moved things into a partial and forgot to move thee translation keys, thanks for the heads up 
there!
- I added the content class and forgot to move the darker-background class.
- I moved the teal links to the darker-background class :+1: 
- I would never know that trick with the opacity to show the shadow above :ok_hand: 

I replied to Pau's comments in the threads above.

This is ready for review again.

luisramos0 commented:
Sorry for making this PR a bit too long but I just added one extra commit (the last one) to close #4774 here as well. I did this because there are dependencies in the code and also because I think it will be good to test these issues together.

sauloperez commented on spec/features/consumer/shopping/shopping_spec.rb:
```diff
@@ -63,8 +64,10 @@
+
           expect(page).to have_selector "option", text: 'frogs'
           expect(page).to have_selector "option", text: 'turtles'
+          expect(page).to have_content "choose when you want your order"
```
shouldn't we use the I18n key to make this a bit less brittle?

luisramos0 replied:
yeah, but there was some problem with the html translation I couldnt fix...

sauloperez commented:
Sorry, guys I see this solves #4772 but that issue is tested (https://github.com/openfoodfoundation/openfoodnetwork/issues/5216#issuecomment-624558857) but this PR is still in Test Ready.  Shall I merge?

daniellemoorhead commented:
#4573 ACs not met, see that issue for details.

daniellemoorhead commented:
@sauloperez I was in the middle of testing, thus some being in ready to go and others not. Changes are required so I'll close those issues that have been done correctly and leave this with the one that remains.

luisramos0 commented:
ok @daniellemoorhead thanks for testing, I'll have a look at the feedback in #4573

luisramos0 commented:
I rebased to resolved a simple conflict on a spec.
